roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,137 @@
+
+        return {"changed": True}
+
+    def get_required_images(self, is_containerized, reg_prefix, openshift_release):
+        if is_containerized:
+            return set(self.containerized_docker_images(reg_prefix, openshift_release))
+
+        return set(self.rpm_docker_images(reg_prefix.split("/")[0], openshift_release))
+
+    @staticmethod
+    def containerized_docker_images(openshift_prefix, version):
+        return [
+            "{openshift_prefix}:{version}".format(openshift_prefix=openshift_prefix, version=version)
+        ]
+
+    @staticmethod
+    def rpm_docker_images(openshift_prefix, version):
+        return [
+            "{openshift_prefix}/registry-console:{version}".format(openshift_prefix=openshift_prefix, version=version)
+        ]
+
+    @staticmethod
+    def qualified_docker_images(openshift_prefix, version):
+        return [
+            "{openshift_prefix}-{component}:{version}".format(**locals())
+            for component in "haproxy-router docker-registry deployer pod".split()
+        ]
+
+    def inspect_images(self, registry, tagged_images, task_vars):
+        failed = set()
+        for image in tagged_images:
+            # the run command pulls the {skopeo_image} if it has not already been pulled.
+            # if pulling this image fails, the current image will be added to the failed set.
+            cmd_str = "docker run {skopeo_image} skopeo inspect docker://{registry}/{tagged_image}".format(
+                skopeo_image=self.skopeo_image,
+                registry=registry,
+                tagged_image=image
+            )
+
+            # use a "docker run ..." command on the remote host to inspect each specific
+            # image inside of a Skopeo image container. If the command's return code is
+            # non-zero, it is safe to assume that an image with the specified tag does not
+            # exist in the given registry.
+            if self.exec_cmd(cmd_str, task_vars)["rc"] != 0:
+                failed.add(image)
+
+        return failed
+
+    def inspect_local_images(self, tagged_images, task_vars):
+        failed = set()
+        for image in tagged_images:
+            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+            if not cmd.get("images"):
+                failed.add(image)
+
+        return failed
+
+    # ensures that the skopeo docker image exists, or can be
+    # pulled, and that the skopeo binary can be executed in it.
+    def check_skopeo_image(self, task_vars):
+        cmd_str = "docker run {skopeo_image} skopeo --version".format(
+            skopeo_image=self.skopeo_image,
+        )
+
+        return self.exec_cmd(cmd_str, task_vars)["stderr"]
+
+    def get_docker_info(self, task_vars):
+        return self.module_executor("docker_info", {}, task_vars)["info"]
+
+    def exec_cmd(self, cmd_str, task_vars):
+        return self.module_executor("command", {"_raw_params": cmd_str}, task_vars)
```

Comment:
How about we vendor the fixed module for the time being, and we can drop it once we require Ansible 2.3? That way our health check wouldn't need to change.